### PR TITLE
feat(driving-learners-permit): Application initialized.

### PR DESCRIPTION
**Pull request by kksteini**
[Screencast from þri 25.apr 2023 09:45:03.webm](https://user-images.githubusercontent.com/77672665/234239697-417bcae7-14f2-45c5-a86e-f53695b5d0ad.webm)

## Checklist:

- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] Formatting passes locally with my changes
- [X] I have rebased against main before asking for a review

### CODE REVIEW

**Review by saevarma — APPROVED** [review]
Core files LGTM

File: libs/api/domains/driving-license-book/src/drivinLicenceBook.type.ts [hunk 1]
```diff
@@ -106,6 +107,10 @@ export interface Organization {
   allowedDrivingSchoolTypes: SchoolType[]
 }
 
+export interface StudentMentorability {
+  eligible: boolean
+}
+
```
Comment (kksteini) [inline libs/api/domains/driving-license-book/src/drivinLicenceBook.type.ts:113]: Is this used?

File: libs/api/domains/driving-license-book/src/drivinLicenceBook.type.ts [hunk 2]
```diff
@@ -75,6 +75,7 @@ export interface DrivingLicenseBook {
   teachersAndLessons: DrivingBookLesson[]
   drivingSchoolExams: DrivingSchoolExam[]
   testResults: DrivingLicenceTestResult[]
+  practiceDriving?: boolean
```
Comment (kksteini) [inline libs/api/domains/driving-license-book/src/drivinLicenceBook.type.ts:78]: Is this used?

File: libs/application/template-api-modules/src/lib/modules/templates/driving-learners-permit/driving-learners-permit.service.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    super(ApplicationTypes.DRIVING_LEARNERS_PERMIT)
+  }
+
+  async completeApplication({
+    application,
+    auth,
+  }: TemplateApiModuleActionProps) {
+    const practicePermitApplication = await this.drivingLicenseService.postPracticePermitApplication(
+      {
+        token: auth.authorization.split(' ')[1],
```
Comment (kksteini) [inline libs/application/template-api-modules/src/lib/modules/templates/driving-learners-permit/driving-learners-permit.service.ts:25]: Mention that this is done in order to get rid of the 'Bearer ' part

File: libs/application/templates/driving-learners-permit/src/lib/messages.ts [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+    id: 'dlp.application:introSectionTitle',
+    defaultMessage: 'Varðandi umsókn',
+    description: 'Title for intro section',
+  },
+  introSectionDescription: {
+    id: 'dlp.application:introSectionDescription',
+    defaultMessage:
+      'Þú ert að fara að sækja um réttindi til þess að vera leiðbeinandi fyrir æfingarakstur. Þú þarft að fylla út kennitölu nemanda sem þú vilt vera leiðbeinandi fyrir. Eftir það verða upplýsingar sóttar úr Ökuskírteinaskrá og úrskurðað um hvort þú hafir tilskilin réttindi til þess að gerast leiðbeinandi fyrir valinn nemanda.',
+    description: 'Description for intro section',
+  },
```
Comment (kksteini) [inline libs/application/templates/driving-learners-permit/src/lib/messages.ts:30]: Text not used anywhere?

File: libs/api/domains/driving-license-book/src/drivingLicenseBook.resolver.ts [hunk 5]
```diff
@@ -145,6 +147,11 @@ export class DrivingLicenseBookResolver {
     return this.drivingLicenseBookService.getSchoolTypes()
   }
 
+  @Query(() => StudentMentorability)
+  studentMentorability(@Args('input') input: StudentMentorabilityInput) {
+    return this.drivingLicenseBookService.getStudentMentorability(input)
+  }
+
```
[the remark this answers is not in the document]
Comment (kksteini) [reply libs/api/domains/driving-license-book/src/drivingLicenseBook.resolver.ts]: Nope, deleted

File: libs/application/template-api-modules/src/lib/modules/templates/driving-learners-permit/driving-learners-permit.service.ts (hunk 3 above)
Comment (kksteini) [reply libs/application/template-api-modules/src/lib/modules/templates/driving-learners-permit/driving-learners-permit.service.ts]: :+1:

**Review by albinagu — APPROVED** [review]
💅 ✅

**Comment by Thorrik58** [comment]
image
The colors are off, is this intended?

File: libs/application/templates/driving-learners-permit/src/fields/LookupStudent/index.tsx [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+  const [getIdentity, { loading: identityQueryLoading }] = useLazyQuery<
+    Query,
+    { input: IdentityInput }
+  >(IDENTITY_QUERY, {
+    onError: (error: unknown) => {
+      setError(fieldNames.lookupError, {
+        type: 'serverError',
+        message: m.errorNationalIdNoName.defaultMessage,
+      })
+      console.log('getIdentity error:', error)
```
Comment (Thorrik58) [inline libs/application/templates/driving-learners-permit/src/fields/LookupStudent/index.tsx:52]: Console.log

**Comment by kksteini** [comment]
Can't reply directly to the colour comment so I'm writing this here.
If the colours are off (I'm colourblind and have a filter maybe that affects the recording) we will have to hotfix it.
This application won't be opened to the public just yet and we have time to look at the colours if need be.

File: libs/application/templates/driving-learners-permit/src/fields/LookupStudent/index.tsx [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+            backgroundColor="blue"
+            loading={identityQueryLoading}
+          />
+        </GridColumn>
+        <GridColumn span="6/12">
+          <InputController
+            id={fieldNames.studentName}
+            name={fieldNames.studentName}
+            label={formatMessage(m.studentNameLabel)}
+            readOnly
```
Comment (Thorrik58) [inline libs/application/templates/driving-learners-permit/src/fields/LookupStudent/index.tsx:169]: Should this field be disabled? The styling suggests it should be filled in by the user.

File: libs/application/templates/driving-learners-permit/src/fields/LookupStudent/index.tsx [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+            name={fieldNames.studentIsMentorable}
+            readOnly
+            defaultValue="default"
+          />
+        </Box>
+      </GridRow>
+
+      <GridRow marginTop={5}>
+        <GridColumn span="12/12">
+          {studentIsMentorable !== 'default' && (
```
Comment (Thorrik58) [inline libs/application/templates/driving-learners-permit/src/fields/LookupStudent/index.tsx:185]: The variable name suggest a boolean value, should this be renamed to something else?

File: libs/application/templates/driving-learners-permit/src/lib/DrivingLearnersPermitTemplate.ts [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+  ApplicationContext,
+  ApplicationStateSchema<DrivingLearnersPermitTemplateEvent>,
+  DrivingLearnersPermitTemplateEvent
+> = {
+  type: ApplicationTypes.DRIVING_LEARNERS_PERMIT,
+  name: m.name,
+  institution: m.institutionName,
+  dataSchema: dataSchema,
+  featureFlag: Features.drivingLearnersPermit,
+  readyForProduction: true,
```
Comment (Thorrik58) [inline libs/application/templates/driving-learners-permit/src/lib/DrivingLearnersPermitTemplate.ts:30]: Not needed anymore, please remove

**Review by Thorrik58 — APPROVED** [review]
A couple of comments I would normally want addressed before approving but as its blocking the release I expect them to be fixed later on.
Could we try to get these bigger changes to review earlier than the day of release or the day before?
Also I wouldnt mind getting some more context in the pull request template instead of removing all the info. The recording was nice tho.

File: libs/application/templates/driving-learners-permit/src/fields/LookupStudent/index.tsx (hunk 7 above)
Comment (albinagu) [reply libs/application/templates/driving-learners-permit/src/fields/LookupStudent/index.tsx:169]: the nationalid is filled in and name is autofilled

File: libs/application/templates/driving-learners-permit/src/fields/LookupStudent/index.tsx (hunk 8 above)
Comment (albinagu) [reply libs/application/templates/driving-learners-permit/src/fields/LookupStudent/index.tsx:185]: yes, probably

**Pull request merged**